enterprise/backend/src/metabase_enterprise/metabot_v3/tools/filters.clj

```diff
@@ -9,11 +9,61 @@
    [metabase.lib.util :as lib.util]
    [metabase.util :as u]))
 
+(defn- apply-filter-bucket
+  [column bucket]
+  (case bucket
+    :second-of-minute (lib/get-second column)
+    :minute-of-hour   (lib/get-minute column)
+    :hour-of-day      (lib/get-hour column)
+    :day-of-week      (lib/get-day-of-week column :iso)
+    :day-of-month     (lib/get-day column)
+    :week-of-year     (lib/get-week column :iso)
+    :month-of-year    (lib/get-month column)
+    :quarter-of-year  (lib/get-quarter column)
+    :year-of-era      (lib/get-year column)
+    ;; these below work in queries but not in the UI
+    #_:millisecond
+    #_:second
+    #_:minute
+    #_:hour
+    #_:day
+    #_:week
+    #_:month
+    #_:quarter
+    #_:year
+    #_:day-of-year
```

Comment:
I think if truncating doesn't work for filters in the UI is not a huge blocker. We likely would not need this too often.
From what I recall, you can compare a timestamp against a date value anyways, right? So if I have a timestamp column I want to filter on I could say `created_at > '2024-01-01'` for example and it should work. So I would not need to truncate `created_at`